full-service/src/service/txo.rs

```diff
@@ -213,17 +234,21 @@ where
             ))
         }
 
-        Ok(self.build_transaction(
+        let (unsigned_tx, fog_resolver) = self.build_unsigned_transaction(
             &account_id_hex,
             &addresses_and_amounts,
             Some(&[txo_id.to_string()].to_vec()),
             fee_value,
             fee_token_id,
             tombstone_block,
             None,
-            None,
             TransactionMemo::RTH,
-        )?)
+        )?;
+
+        let account = Account::get(&AccountID(account_id_hex.clone()), &conn)?;
+        let account_key: AccountKey = mc_util_serial::decode(&account.account_key)?;
+
+        Ok(unsigned_tx.sign(&account_key, fog_resolver)?)
```

Comment:
I've seen these few lines of getting an ccount, decoding the key and then signing repeated a few times. Might be worth refactoring into a helper method.